components/Comment/CommentSidebarToggle.tsx

```diff
@@ -3,39 +3,76 @@ import colors from "./lib/colors";
 import { faComments } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import ResearchCoinIcon from "../Icons/ResearchCoinIcon";
+import { createPortal } from "react-dom";
+import { useEffect, useState } from "react";
+import config from "./lib/config";
 
 type Args = {
   setIsOpen: Function;
   isOpen: boolean;
   bountyAmount: number;
   commentCount: number;
-  isReady: boolean;
 };
 
-const CommentSidebarToggle = ({ isOpen, setIsOpen, isReady, bountyAmount = 0, commentCount = 0 }: Args) => {
+const CommentSidebarToggle = ({
+  isOpen,
+  setIsOpen,
+  bountyAmount = 0,
+  commentCount = 0,
+}: Args) => {
   const hasActiveBounties = bountyAmount > 0;
-  const bountyDisplayVal = (bountyAmount > 1000 ? bountyAmount / 1000 : bountyAmount).toFixed(0);
+  const bountyDisplayVal =
+    bountyAmount > 1000
+      ? (bountyAmount / 1000).toFixed(0) + "k"
+      : bountyAmount.toFixed(0);
 
-  if (!isReady) {
-    return null;
-  }
+  const [mountEl, setMountEl] = useState<HTMLElement | null>(null);
+  useEffect(() => {
+    if (!mountEl) {
+      const _mountEl = document.getElementById(config.toggle.elemToMountAt);
```

Comment:
This feels like a weird way to write react. Is there a way we can do this without trying to get the element by Id and then conditionally calling createPortal?